src/dyncfg/src/lib.rs

```diff
@@ -217,6 +206,64 @@ impl ConfigSet {
     pub fn entries(&self) -> impl Iterator<Item = &ConfigEntry> {
         self.configs.values()
     }
+
+    /// Executes `f` with updates to configuration values frozen.
+    ///
+    /// This allows `f` to observe multiple configuration values atomically.
+    /// Specifically, updates from [`ConfigUpdates`] will never be partially
+    /// applied when `f` runs: either all updates from the batch will be applied
+    /// or none of them will.
+    ///
+    /// `f` should take care to execute quickly, as no further configuration
+    /// updates can be applied while `f` is executing.
+    pub fn with_updates_frozen<F>(&self, mut f: F)
```

Comment:
This isn't presently used, but it seemed like it could be useful in the future!

Reply:
it DOES seem useful, for some storage stuff